scripts/check-go-version.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+declare -A go_version_map
+version_regex="[0-9]\.[0-9]"
+
+# Fetch latest go release version
+go_latest_version=$(curl -s https://go.dev/dl/?mode=json | jq -r '.[0].version' | awk -F'.' '{gsub("go", ""); print $1"."$2}')
+
+# Extract go version from go.mod
+go_mod_path="./go.mod"
+go_mod_version=$(grep -oP "go\s+\K$version_regex+" $go_mod_path)
+
+# Ensure go.mod uses go_latest_version - 1
+if [ "${go_mod_version%.*}" != "${go_latest_version%.*}" ] || [ $((10#${go_latest_version#*.} - 10#${go_mod_version#*.})) != 1 ]; then
```

Comment:
I don't think this is what we want. With this check the CI will start failing immediately after Go release, which is unfair to PR authors. My intention was that the script checks that all other build scripts are on some version N and the go.mod is on N-1, but N does not have to be latest. Latest can be printed, but not enforced. It should only be used in the update script.